docs/guides/backend_guide.md

```diff
@@ -0,0 +1,187 @@
+------------------------+-----------+------------+------------+------------+----------+-------------+-------------
+ terraform_remote_state | states    | my_initial_user |            | t          | f        | f           | f
+```
+
+After running terraform `apply`, to terraform.tfstate on the database will look something like this:
+
+```shell
+rdb=> SELECT * FROM information_schema.columns
+WHERE table_schema = 'terraform_remote_state'
+AND TABLE_NAME = 'states';
+ id |  name   | data
+----+---------+----------------------------------------------------------
+  1 | default | {                                                       +
+    |         |   "version": 4,                                         +
+    |         |   "serial": 0,                                          +
+    |         |   "lineage": "07a1e05b-3cba-438a-0c70-3ec5e73d4baf",    +
+    |         |   "outputs": {},                                        +
+    |         |   "resources": [                                        +
+    |         |     {
+    
+    ....
+```
+
+## Multiple Workplaces
+
+You can configure several `states` on your database using a different `schema_name`.
+
+Then workspaces are appended to that key to generate a separate state for each workspace.
+Since tracking of the workspaces is in the table inside PostgreSQL, we need to separate the different states we want to track.
+
+We can do that in one of two ways: separate databases or separate schemas.
+
+```hcl
+terraform {
+  # Omitted
+
+  backend "pg" {
+    schema_name = "other_state"
+  }
+}
+```
+
+## Migrating the state
+
+Considering you have already running infrastructure you want to use the `backend` option.
+
+All we need to do is initialize Terraform passing the backend configuration.
+
+Terraform should ask if you want to migrate from local to the new remote backend.
+
+Answer the prompt `yes`, and your state will migrate.
+
+```shell
+$ terraform init -backend-config="conn_str=${PG_CONN_STR}" -migrate-state
+```
+
+## What about locking?
+
+Most of the remote backends natively support locking. To run terraform apply, Terraform will automatically acquire a lock;
+if someone else is already running apply, they will already have the lock, and you will have to wait.
+You can run apply with the `-lock-timeout=<TIME>` parameter to tell Terraform to wait up to TIME for a lock to be released (e.g., `-lock-timeout=10m` will wait for 10 minutes).
+
+The Lock method, prevents opening the state file while already in use.
+
+## Share configuration
+
+You can also share the configuration using the different [data sources](https://www.terraform.io/language/state/remote-state-data).
+this is useful when you are working on the same infrastructure we the same team.
+
+For further information about thea advantages of t
+
+```hcl
+data "scaleway_rdb_instance" "mybackend" {
+    name = "your-database-name"
+}
+```
+
+## Limitation
+The Scaleway Object Storage based backends is not supported by scaleway.
```

Comment:
This sentence needs to be reworked. Only locking is not supported, but storing of the state is